Frontend/CO.CDP.OrganisationApp/Pages/Forms/_FormElementUrlInput.cshtml

```diff
@@ -0,0 +1,88 @@
+@model FormElementUrlInputModel
+
+@{
+    var textInputHasError = ((TagBuilder)Html.ValidationMessageFor(m => m.TextInput)).HasInnerHtml;
+    var selectOptionHasError = ((TagBuilder)Html.ValidationMessageFor(m => m.HasValue)).HasInnerHtml;
+    string textInputAriaDescribedby = "";
+
+    void RenderField()
+    {
+        if (textInputHasError)
+        {
+            textInputAriaDescribedby += " TextInput-error";
+
+            <p class="govuk-error-message" id="TextInput-error">
+                <span class="govuk-visually-hidden">Error:</span>
+                @Html.ValidationMessageFor(m => m.TextInput)
+            </p>
+        }
+
+        <input class="govuk-input govuk-!-width-two-thirds" id="TextInput" name="TextInput" type="text" value="@Model.TextInput"
+        @(string.IsNullOrWhiteSpace(textInputAriaDescribedby) ? "" : Html.Raw($"aria-describedby=\"{textInputAriaDescribedby}\""))>
+    }
+}
+
+<div class="govuk-form-group @((Model.IsRequired && textInputHasError) || selectOptionHasError ? "govuk-form-group--error" : "")">
+
+    @if (!string.IsNullOrWhiteSpace(Model.Heading))
+    {
+        <h1 class="govuk-label-wrapper">
+            <label class="govuk-label govuk-label--l" for="TextInput">@Model.Heading</label>
+
+            @if (!string.IsNullOrWhiteSpace(Model.Caption))
+            {
+                <span class="govuk-caption-l govuk-!-margin-bottom-3">@Model.Caption</span>
+            }
+        </h1>
+    }
+
+    @if (!string.IsNullOrWhiteSpace(Model.Description))
+    {
+        textInputAriaDescribedby += "TextInput-description";
+
+        <div id="TextInput-description">
+            @Html.Raw(Model.Description)
+        </div>
+    }
+
+    @if (selectOptionHasError)
+    {
+        <p class="govuk-error-message">
+            <span class="govuk-visually-hidden">Error:</span>
+            @Html.ValidationMessageFor(m => m.HasValue)
+        </p>
+    }
+
+    @if (Model.IsRequired)
+    {
+        RenderField();
+    }
+    else
+    {
+        <fieldset class="govuk-fieldset">
```

Comment:
Fieldsets must have legends. The pattern you've got is correct for the text input only version. But when you've got this radio field combined with the conditional input field, the h1 and legend would need to come inside the fieldset, and the text field itself would have a label separately from the h1 (It could be visually hidden if necessary for the designs)